### PR TITLE
Sanitise relative paths

### PR DESCRIPTION
**Pull request by oerc0122**
See #821 for detailed info

Adds 3 global variables `herbert_path`, `horace_path` and `root_path` which point to relevant parts of the Horace directory structure for internal reference and consistency. 

These are accessible via the `horace_paths` class (in `herbert_core`) there are also methods to get the folder of a function and clearing `paths` as well as the base paths (admin, tests, lowlevelcode) and two commonly used paths (test/common_data, test/common_functions)

These are set in `herbert_init` and `horace_init` or through the `paths` class and cleared in `horace_off`.

Also fixes typo on startup banner.

### CODE REVIEW

**Review by mducle — APPROVED** [review]
Thanks for changing it to a class!

File: horace_core/horace_off.m [hunk 1]
```diff
@@ -7,22 +7,26 @@
 % T.G.Perring
 
 % root directory is assumed to be that in which this function resides
-rootpath = fileparts(fileparts(which('horace_init')));
+global horace_path
```
Comment (mducle) [inline horace_core/horace_off.m:10]: Oh actually... should this not use the `paths` class?

Comment (oerc0122) [reply horace_core/horace_off.m]: Done.

**Review by mducle — APPROVED** [review]

**Review by abuts — APPROVED** [review]
well, do not mind this, though not see big benefit. But let it go -- why not. 
Two notes. 
1) `paths` class sounds very similar to system `path` -- I would rename it to `hor_paths` to avoid confusion or somebody in a future in a toolbox use similar name. 

2) It looks like this does not need a unit test, but I would suggest opposite. Its not difficult to add one

**Pull request merged**